src/common-utils/main.sh

```diff
@@ -437,6 +437,14 @@ if [ "${INSTALL_ZSH}" = "true" ]; then
     fi
 
     if [ "${CONFIGURE_ZSH_AS_DEFAULT_SHELL}" == "true" ]; then
+        # Fixing chsh always asking for a password on alpine linux
+        # ref: https://askubuntu.com/questions/812420/chsh-always-asking-a-password-and-get-pam-authentication-failure.
+        if [ -f "/etc/pam.d/chsh" ] && grep -vEq "^auth[[:blank:]]+sufficient[[:blank:]]+pam_rootok\.so$"; then
+            awk '/^auth(.*)pam_rootok\.so$/ { $2 = "sufficient" } { print }' /etc/pam.d/chsh > /tmp/chsh.tmp && mv /tmp/chsh.tmp /etc/pam.d/chsh
+        else
+            echo "auth sufficient pam_rootok.so" >> /etc/pam.d/chsh
+        fi
+
         chsh --shell /bin/zsh ${USERNAME}
```

Comment:
```suggestion
        CHSH_FILE_EXISTS="false"
        [ -f /etc/pam.d/chsh ] && CHSH_FILE_EXISTS="true"

        echo "auth sufficient pam_rootok.so" >> /etc/pam.d/chsh

        chsh --shell /bin/zsh ${USERNAME}

        if [ "${CHSH_FILE_EXISTS}" = "true" ]; then
            head -n -1 /etc/pam.d/chsh > /tmp/chsh-temp && mv /tmp/chsh-temp /etc/pam.d/chsh
        else
            rm -rf /etc/pam.d/chsh
        fi
```

@krikchaip what do you think of this approach? We are adding `auth sufficient pam_rootok.so`, switching the shell and reverting the changes. 

This way, we achieve our end goal of using `zsh` as default and don't have any lingering changes.

Reply:
You are basically trying to append the `auth sufficient pam_rootok.so` to the file right away. What will happen if there's already a line with `auth required pam_rootok.so`? Which one will take precedence? And the if statement at the bottom doesn't seem right to me.